webapp/src/libs/hooks/useChat.ts

```diff
@@ -355,6 +359,32 @@ export const useChat = () => {
         }
     };
 
+    const deleteChat = async (chatId: string) => {
+        const friendlyChatName = getFriendlyChatName(conversations[chatId]);
+        await chatService
+            .deleteChatAsync(chatId, loggedInUser.id, await AuthHelper.getSKaaSAccessToken(instance, inProgress))
+            .then(() => {
+                dispatch(deleteConversation(chatId));
+
+                // If there is only one chat left, create a new chat
+                if (Object.keys(conversations).length <= 1) {
```

Comment:
This is more of question: will `conversations` get updated after the `dispatch` call?

Reply:
yeah state will automatically get updated if any of its dependencies change, this is handled by redux

Reply:
If that's the case, why `<= 1`? Doesn't that mean it may still have 1 chat?

Reply:
Oh sorry, I was wrong before. Conversations is not automatically updated; if I change the logic to `                if (Object.keys(conversations).length == 0) {` then it doesn't automatically create a new chat if the last one is updated. The method isn't updated, so it maintains the same execution thread so it's using the same conversations object.

Reply:
Got it!